### PR TITLE
chore(docs): add core & middleware

### PR DESCRIPTION
**Pull request by rozakbuhari**
## Description

- Add core initial documentation
- Middleware documentation

### CODE REVIEW

**Comment by rozakbuhari** [comment]
Sure @arrizalamin, HOC looks good, one question:
Can we use the middleware like this?

```ts
const batiq = createBatiq(
  log(multiplayer(persist(appJson)))
);
```

File: docs/middleware.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+```
+
+### `undoRedoMiddleware`
+
+This middleware adds undo/redo functionality.
+
+```ts
+import { undoRedoMiddleware } from "@batiq/core";
+
+const app = createBatiq(appJson, [undoRedoMiddleware]);
```
[the remark this answers is not in the document]
Comment (rozakbuhari) [reply docs/middleware.md:44]: That is make sense! 👍🏻

**Comment by arrizalamin** [comment]
@rozakbuhari yes, current api also possible with HOF
```typescript
const batiq = createBatiq(appJson, [log, multiplayer, storage])
```

**Comment by rozakbuhari** [comment]
OK, let's go with this.

File: docs/middleware.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  log: () => void;
+}
+
+const myMiddleware: Middleware<MyMiddleware> = (batiq: Batiq) => {
+  if (operation.type === "insert") {
+    // Do something
+  }
+
+  // Call the next middleware
+  return next(operation);
```
Comment (arrizalamin) [inline docs/middleware.md:95]: this middleware body should be:
```typescript
const {dispatch} = batiq;

return {
  ...batiq,
  dispatch: (operation) => {
    if (operation.type === "insert") {
      // do something
    }

    return dispatch(operation);
  }
}
```

File: docs/core.md [hunk 3]
```diff
@@ -94,16 +95,16 @@ const appJson: AppSchema = {
   pages: [ ... ],
 };
 
-const app = createBatiq(appJson, [loggerMiddleware, multiplayerMiddleware]);
+const app = createBatiq(log(multiplier(undoRedo(appJson))));
```
Comment (arrizalamin) [inline docs/core.md:98]: we should't change the api, appJson should be passed as a parameter to createBatiq

**Review by arrizalamin — APPROVED** [review]
lgtm

**Pull request merged**